curriculum/challenges/english/25-front-end-development/lab-rpg-character/67d83df6f82eda3868dd2a84.md

```diff
@@ -0,0 +1,192 @@
+
+assert create_character(1, 1, 1, 1) == 'The character name should be a string'
+`
+    )
+})
+```
+
+When `create_character` is called with a first argument that contains a space it should return `Character name should not contain spaces`.
+
+```js
+({
+    test: () => runPython(
+        `
+assert create_character("cha cha", 4, 1, 2) == 'Character name should not contain spaces'
+`
+    )
+})
+```
+
+When `create_character` is called with a first argument that is longer than 10 characters it should return `Character name is too long`.
+
+```js
+({
+    test: () => runPython(
+        `
+assert create_character("aaabbbcccddd", 4, 2, 1) == 'Character name is too long'
+`
+    )
+})
+```
+
+When `create_character` is called with a second, third or fourth argument that is not an integer it should return `All stats should be integers`.
+
+```js
+({
+    test: () => runPython(
+        `
+wrong_types = ["3", "5", {"a": 3}, 3.1, [1, 2], (1, 2), {1, 2}, "str", "friend"]
+for wrong_type in wrong_types:
+    assert create_character("friend", wrong_type, 2, 1) == 'All stats should be integers'
+    assert create_character("friend", 2, wrong_type, 1) == 'All stats should be integers'
+    assert create_character("friend", 2, 1, wrong_type) == 'All stats should be integers'
+`
+    )
+})
+```
+
+When `create_character` is called with a second, third or fourth argument that is lower than `1` it should return `All stats should be a minimum of 1`.
+
+```js
+({
+    test: () => runPython(
+        `
+assert create_character("ren", 0, 4, 3) == 'All stats should be a minimum of 1'
+assert create_character("ren", 4, 0, 3) == 'All stats should be a minimum of 1'
+assert create_character("ren", 4, 3, 0) == 'All stats should be a minimum of 1'
+`
+    )
+})
+```
+
+When `create_character` is called with a second, third or fourth argument that is higher than `4` it should return `All stats should be no more than 4`.
+
+```js
+({
+    test: () => runPython(
+        `
+assert create_character("ren", 1, 5, 1) == 'All stats should be no more than 4'
+assert create_character("ren", 5, 1, 1) == 'All stats should be no more than 4'
+assert create_character("ren", 1, 1, 5) == 'All stats should be no more than 4'
+`
+    )
+})
+```
+
+When `create_character` is called with a second, third or fourth argument that do not sum to `7` it should return `The character should start with 7 points`.
+
+```js
+({
+    test: () => runPython(
+        `
+assert create_character("ren", 4, 4, 4) == 'The character should start with 7 points'
+assert create_character("ren", 1, 1, 1) == 'The character should start with 7 points'
+assert create_character("ren", 1, 2, 3) == 'The character should start with 7 points'
+`
+    )
+})
+```
+
+`create_character("ren", 4, 2, 1)` should return `ren\nSTR ●●●●○○○○○○\nINT ●●○○○○○○○○\nCHA ●○○○○○○○○○`.
+
+```js
+({
+    test: () => runPython(
+        `
+assert create_character("ren", 4, 2, 1) == f'ren\\nSTR ●●●●○○○○○○\\nINT ●●○○○○○○○○\\nCHA ●○○○○○○○○○'
```

Comment:
```suggestion
assert create_character("ren", 4, 2, 1) == f'ren\\nSTR ●●●●○○○○○○\\nINT ●●○○○○○○○○\\nCHA ●○○○○○○○○○'
assert create_character("Bob", 1, 2, 4) == f'Bob\\nSTR ●○○○○○○○○○\\nINT ●●○○○○○○○○\\nCHA ●●●●○○○○○○'
```
I would add another check to avoid hardcoding.